go/enclave/evm/evm_facade.go

```diff
@@ -40,269 +37,140 @@ const (
-			statedb.AddBalance(header.Coinbase, uint256.MustFromBig(executionGasCost), tracing.BalanceDecreaseGasBuy)
-		}
-		receipt.GasUsed += l1Gas.Uint64()
-
-		return receipt, err
-	}
-
-	receipt, err := applyTx(cc, chain, nil, gp, s, header, tx, usedGas, vmCfg)
-	// adjust the receipt to point to the right batch hash
-	if receipt != nil {
-		receipt.Logs = s.GetLogs(tx.Tx.Hash(), header.Number.Uint64(), header.Hash())
-	}
+	snap := s.Snapshot()
+	res, err := exec.execute(tx, from, s, header, gp, usedGas, tCount, noBaseFee)
 	if err != nil {
 		s.RevertToSnapshot(snap)
 		return &core.TxExecResult{
-			Receipt:      receipt,
 			TxWithSender: &core.TxWithSender{Tx: tx.Tx, Sender: &from},
 			Err:          err,
 		}
 	}
-
-	contractsWithVisibility := make(map[gethcommon.Address]*core.ContractVisibilityConfig)
-	for _, contractAddress := range createdContracts {
-		contractsWithVisibility[*contractAddress] = readVisibilityConfig(vmenv, contractAddress)
-	}
-
-	return &core.TxExecResult{
-		Receipt:          receipt,
-		TxWithSender:     &core.TxWithSender{Tx: tx.Tx, Sender: &from},
-		CreatedContracts: contractsWithVisibility,
-	}
+	return res
 }
 
-const (
-	maxGasForVisibility = 30_000 // hardcode at 30k gas.
-)
-
-func readVisibilityConfig(vmenv *vm.EVM, contractAddress *gethcommon.Address) *core.ContractVisibilityConfig {
-	cc, err := NewTransparencyConfigCaller(*contractAddress, &localContractCaller{evm: vmenv, maxGasForVisibility: maxGasForVisibility})
-	if err != nil {
-		// unrecoverable error. should not happen
-		panic(fmt.Sprintf("could not create transparency config caller. %v", err))
-	}
-	visibilityRules, err := cc.VisibilityRules(nil)
-	if err != nil {
-		// there is no visibility defined, so we return auto
-		return &core.ContractVisibilityConfig{AutoConfig: true}
+func (exec *evmExecutor) execute(tx *common.L2PricedTransaction, from gethcommon.Address, s *state.StateDB, header *types.Header, gp *gethcore.GasPool, usedGas *uint64, tCount int, noBaseFee bool) (*core.TxExecResult, error) {
+	// a transaction can create multiple contracts.
+	// we use a tracer hook to collect the addresses
+	var createdContracts []*gethcommon.Address
+	cfg := vm.Config{
+		NoBaseFee: noBaseFee,
+		Tracer: &tracing.Hooks{
+			// called when the code of a contract changes.
+			OnCodeChange: func(addr gethcommon.Address, prevCodeHash gethcommon.Hash, prevCode []byte, codeHash gethcommon.Hash, code []byte) {
+				// only proceed for new deployments.
+				if len(prevCode) > 0 {
+					exec.logger.Debug("OnCodeChange: Skipping contract deployment", "address", addr.Hex())
+					return
+				}
+				createdContracts = append(createdContracts, &addr)
+				exec.logger.Debug("OnCodeChange: Contract deployed", "address", addr.Hex())
+			},
+		},
 	}
 
-	transp := false
-	if visibilityRules.ContractCfg == transparent {
-		transp = true
-	}
+	blockContext := gethcore.NewEVMBlockContext(header, exec.chain, nil)
+	// todo - why do we need the blobhashes here?
```

Comment:
I have absolutely no idea. If people are submitting blobs, I guess they need to be given to the evm context so blobhash opcode returns something meaningful

Reply:
we don't support blob transactions on the l2